### ISSUE TITLE
No meaningful way to use issue_client_certificate = true?

### ISSUE THREAD

**Issue by DoomGerbil** [terraform-google-modules/terraform-google-kubernetes-engine#1299]
### TL;DR

If a cluster is created with `issue_client_certificate = true`, the module outputs do not include the `client_certificate` or `client_key` generated at cluster creation time - and there is no other mechanism for retrieving them, short of parsing the tfstate file - both of which are necessary in order to use this auth method.

### Expected behavior

If a cluster is created with `issue_client_certificate = true`, the module outputs should include [`master_auth.0.client_certificate`](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster#master_auth.0.client_certificate) and [`master_auth.0.client_key`](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster#master_auth.0.client_key).

### Observed behavior

These values are not outputted by the module, and there is no supported way to retrieve them.

### Terraform Configuration

```hcl
module "gke" {
  source                     = "terraform-google-modules/kubernetes-engine/google"
  version                    = "v21.1.0"

  project_id                 = var.gcp_project_name
  name                       = var.gke_cluster_name

  regional                   = false
  zones                      = var.zone

  network                    = var.vpc_name
  subnetwork                 = var.subnet_name
  ip_range_pods              = var.pods_ip_range_name
  ip_range_services          = var.services_ip_range_name

  issue_client_certificate = true
}
```

### Terraform Version

```sh
❯ terraform version
Terraform v1.1.6
on darwin_amd64
```

**Comment by bharathkkb** [comment]
@DoomGerbil Thanks for the report. We could add this but considering it is a legacy method I am not sure if we want to add top level outputs for this. Can you use a [data source](https://registry.terraform.io/providers/hashicorp/google/latest/docs/data-sources/container_cluster) instead?

**Comment by DoomGerbil** [comment]
> We could add this but considering it is a legacy method I am not sure if we want to add top level outputs for this.

That's sort of what I expected the answer to be, but figured it was worth a report.

For context, our use case here is that we create short-lived GKE clusters used for practical interview sessions. We need to be able to give interview candidates - who have no association with our organization, and may not have the GCP SDK available on their personal machines - access to the cluster for the lifespan of said cluster. Because of those limitations, we cannot require them to use `gcloud` auth to talk to the cluster, which leaves client cert auth as our best option.

> Can you use a [data source](https://registry.terraform.io/providers/hashicorp/google/latest/docs/data-sources/container_cluster) instead?

I will give this a try. In the immediate term, I have hacked up an ugly shell/jq script that parses the data directly from the local `tfstate` file and renders it into a valid kubeconfig file, which is run by Terraform via `local-exec` upon creation of the cluster.

I don't like that "solution", for obvious reasons, so I will give this a shot. Thanks!

**Comment by bharathkkb** [comment]
Gotcha, closing this out since there is a solution. You may also be interested in https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/tree/master/modules/auth which uses the datasource for kubeconfig generation.
